Nrf profiler timestamp (#28314)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

|no_changes_yet_note|

> DFU librariesgit commit --amend

Reviewer: looks like a typo

Other libraries
---------------

> * :ref:`nrf_profiler` library:

Reviewer 2: Please remove dot at the end of commit message title and align number of characters per line in the commit message body to 72 character limit.

Author: Corrected

  * Added sending :kconfig:option:`CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC` through the info RTT channel as part of the system description.
    This enables support for the nRF54L SoC Series (:kconfig:option:`CONFIG_SOC_SERIES_NRF54L`) by providing accurate hardware clock information to the profiler host tools.
    The hardware clock frequency varies between different SoCs, and host tools require this information to correctly interpret received timestamps.
    This change breaks backward compatibility between the library and host tools due to modifications in the info RTT channel string layout.

>     System event descriptions are now sent within the ``<ev_info_start>`` and ``<ev_info_stop>`` markers.

Reviewer 2: `nRF Profiler event...` ?

    System configuration details are now sent within the ``<sys_config_start>`` and ``<sys_config_stop>`` markers.

>     The system configuration details are formatted as a comma-separated list of key-type-value data, for example: ``key1,u32,value1``.

Reviewer 2: ```suggestion
    The system configuration details are formatted as a list of key-type-value data, for example: ``key1,u32,value1``.
    Subsequent entries are separated by newlines.
```

>     The system configuration details are formatted as a comma-separated list of key-value pairs, for example: ``key1:value1,key2:value2``.

Reviewer 2: Consider using `\n` instead of `,` as a separator (to make approach more consistent with informing about supported profiler events). We might also add extra `\n` to put tags in separate lines.

>     The :kconfig:option:`CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC` parameter is sent using the ``sys_clock_hw_cycles_per_sec`` key.

Reviewer 2: ```
Value returned by the :c:func:`sys_clock_hw_cycles_per_sec` function is sent
```

>     Specifically, the :kconfig:option:'CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC' parameter is sent using the sys_clock_hw_cycles_per_sec key.

Reviewer: ```suggestion

    This change breaks backward compatibility between the nrf_profiler and host tools due to modifications in the info RTT channel string layout.
  System event descriptions are now sent within ``<ev_info_start>`` and ``<ev_info_stop>`` markers.
  System configuration details are now sent within ``<sys_config_start>`` and ``<sys_config_stop>`` markers.
  The system configuration details are formatted as a comma-separated list of key-value pairs, for example: ``key1:value1,key2:value2``.
  The :kconfig:option:`CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC` parameter is sent using the ``sys_clock_hw_cycles_per_sec`` key.
```

Other libraries
---------------

* :ref:`nrf_profiler` library:

  * Added sending :kconfig:option:`CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC` through the info RTT channel as part of the system description.
    This enables support for the nRF54L SoC Series (:kconfig:option:`CONFIG_SOC_SERIES_NRF54L`) by providing accurate hardware clock information to the profiler host tools.
    The hardware clock frequency varies between different SoCs, and host tools require this information to correctly interpret received timestamps.

>     This change breaks backward compatibility between the nrf_profiler and host tools due to modifications in the info RTT channel string layout.

Reviewer 3: ```suggestion
    This change breaks backward compatibility between the library and host tools due to modifications in the info RTT channel string layout.
```

    System event descriptions are now sent within ``<ev_info_start>`` and ``<ev_info_stop>`` markers.

>     System configuration details are now sent within ``<sys_config_start>`` and ``<sys_config_stop>`` markers.

Reviewer 3: ```suggestion
    System event descriptions are now sent within the ``<ev_info_start>`` and ``<ev_info_stop>`` markers.
    System configuration details are now sent within the ``<sys_config_start>`` and ``<sys_config_stop>`` markers.
```

Other libraries
---------------

* :ref:`nrf_profiler` library:

  * Added sending :kconfig:option:`CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC` through the info RTT channel as part of the system description.

>     This enables support for the nRF54L SoC Series (:kconfig:option:`CONFIG_SOC_SERIES_NRF54L`) by providing accurate hardware clock information to the profiler host tools.

Reviewer 3: ```suggestion
    This enables support for the nRF54L Series SoCs (:kconfig:option:`CONFIG_SOC_SERIES_NRF54L`) by providing accurate hardware clock information to the profiler host tools.
```

>     The hardware clock frequency varies between different SoCs, and host tools require this information to correctly interpret received timestamps.

Reviewer 3: ```suggestion
    The hardware clock frequency varies between different SoCs and the host tools require this information to correctly interpret the received timestamps.
```

>     This change breaks backward compatibility between the library and host tools due to modifications in the info RTT channel string layout.

Reviewer 3: ```suggestion
    This change breaks the backward compatibility between the library and host tools due to modifications in the info RTT channel string layout.
```

    The nRF Profiler event descriptions are now sent within the ``<ev_info_start>`` and ``<ev_info_stop>`` markers.
    System configuration details are now sent within the ``<sys_config_start>`` and ``<sys_config_stop>`` markers.

>     The system configuration details are formatted as a list of key-type-value data, for example: ``key1,u32,value1``. Subsequent entries are separated by newlines.

Reviewer 3: ```suggestion
    The system configuration details are formatted as a list of key-type-value data, for example: ``key1,u32,value1``.
    Subsequent entries are separated by newlines.
```

This section provides detailed lists of changes by :ref:`script <scripts>`.

* :ref:`west_sbom` script:

  * Added licence handling of the LLVM toolchain libraries.

* :ref:`nrf_profiler_script` script:

>   * Corrected the ``ms_per_timestamp_tick`` parameter to match the nRF52 system clock.

Reviewer 2: Consider mentioning that it's defined in `rtt_nordic_config.py` file

Reviewer 2: `to accurately match`?

Reviewer 2: Consider also mentioning nRF53 and nRF91 SoC Series (hw cycles per second would match too and they are also supported by profiler)

This section provides detailed lists of changes by :ref:`script <scripts>`.

* :ref:`west_sbom` script:

  * Added licence handling of the LLVM toolchain libraries.

* :ref:`nrf_profiler_script` script:

  * Updated:

>     * Fixed the ``ms_per_timestamp_tick`` parameter (defined in ``rtt_nordic_config.py``) to align with the system clock frequency of nRF52, nRF53, and nRF91 SoCs.

Reviewer 2: ```suggestion
    * Fixed the ``ms_per_timestamp_tick`` parameter (defined in :file:`rtt_nordic_config.py`) to align with the system clock frequency of nRF52, nRF53, and nRF91 SoCs.
```

Reviewer 2: `... to accurately align ...` (with the current wording, I am a bit worried that it sounds a bit like if the value was completely wrong/misaligned)

This section provides detailed lists of changes by :ref:`script <scripts>`.

* :ref:`west_sbom` script:

  * Added licence handling of the LLVM toolchain libraries.

* :ref:`nrf_profiler_script` script:

  * Corrected the ``ms_per_timestamp_tick`` parameter to match the nRF52 system clock.
  * Updated the scripts to use hardware clock information from SoC and to enable accurate conversion from system ticks to seconds.

>     Refactored the parser to support the new RTT communication layout, including the handling of ``<ev_info_...>`` and ``<sys_config_...>`` data markers.

Reviewer 3: ```suggestion
  * Updated:

    * The scripts to use hardware clock information from the SoC and to enable accurate conversion from system ticks to seconds.
    * By refactoring the parser to support the new RTT communication layout, including the handling of ``<ev_info_...>`` and ``<sys_config_...>`` data markers.

  * Fixed the ``ms_per_timestamp_tick`` parameter to match the nRF52 system clock.
```

This section provides detailed lists of changes by :ref:`script <scripts>`.

* :ref:`west_sbom` script:

  * Added licence handling of the LLVM toolchain libraries.

* :ref:`nrf_profiler_script` script:

  * Updated:

    * Fixed the ``ms_per_timestamp_tick`` parameter (defined in ``rtt_nordic_config.py``) to align with the system clock frequency of nRF52, nRF53, and nRF91 SoCs.
    * Updated the scripts to use hardware clock information from SoC and to enable accurate conversion from hardware clock cycles to seconds.

>       Refactored the parser to support the new RTT communication layout, including the handling of ``<ev_info_...>`` and ``<sys_config_...>`` data markers.

Reviewer 3: ```suggestion
  * Updated the scripts to use hardware clock information from the SoC and to enable accurate conversion from hardware clock cycles to seconds.
     Refactored the parser to support the new RTT communication layout, including the handling of ``<ev_info_...>`` and ``<sys_config_...>`` data markers.

  * Fixed the ``ms_per_timestamp_tick`` parameter (defined in the :file:`rtt_nordic_config.py` file) to align with the system clock frequency of nRF52, nRF53, and nRF91 SoCs.
      
```

Author: Corrected

  * Added licence handling of the LLVM toolchain libraries.

* :ref:`nrf_profiler_script` script:

>   * Updated the scripts to use hardware clock information from SoC and make accurate conversion (system ticks to seconds).

Reviewer: ```suggestion
  * Updated the scripts to use hardware clock information from SoC and to enable accurate conversion from system ticks to seconds.
```

>     Refactored the parser to support the new RTT communication layout, including the handling of ``<ev_info_...>`` and ``<sys_config_...>`` data markers.

Reviewer: ```suggestion
  * Refactored the parser to support the new RTT communication layout, including the handling of ``<ev_info_...>`` and ``<sys_config_...>`` data markers.
```

  * Added licence handling of the LLVM toolchain libraries.

* :ref:`nrf_profiler_script` script:

  * Updated the scripts to use hardware clock information from SoC and to enable accurate conversion from system ticks to seconds.

>   * Refactored the parser to support the new RTT communication layout, including the handling of ``<ev_info_...>`` and ``<sys_config_...>`` data markers.

Reviewer 2: Can we put it as one note (in fact it's one change)

Reviewer 2: I would also suggest to add notes on nRF Profiler subsystem and script pages to inform explicitly that NCS 3.4 breaks backwards compatibility between host tools and subsystem (I am worried that release notes might be overlooked).

* :ref:`nrf_profiler_script` script:

  * Corrected the ``ms_per_timestamp_tick`` parameter to match the nRF52 system clock.

>   * Updated the scripts to use hardware clock information from SoC and to enable accurate conversion from system ticks to seconds.

Reviewer 2: ```suggestion
  * Updated the scripts to use hardware clock information from SoC to enable accurate conversion from hardware clock cycles to seconds.
```

Other libraries
---------------

* :ref:`nrf_profiler` library:

  * Added sending :kconfig:option:`CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC` through the info RTT channel as part of the system description.
    This enables support for the nRF54L Series SoCs (:kconfig:option:`CONFIG_SOC_SERIES_NRF54L`) by providing accurate hardware clock information to the profiler host tools.
    The hardware clock frequency varies between different SoCs, and the host tools require this information to correctly interpret the received timestamps.
    This change breaks the backward compatibility between the library and host tools due to modifications in the info RTT channel string layout.
    The nRF Profiler event descriptions are now sent within the ``<ev_info_start>`` and ``<ev_info_stop>`` markers.
    System configuration details are now sent within the ``<sys_config_start>`` and ``<sys_config_stop>`` markers.

>     The system configuration details are formatted as a list of key-value data, for example: ``key1,value1``.

Reviewer 3: ```suggestion
    The system configuration details are formatted as a list of key-value data, for example, ``key1,value1``.
```

Author: Corrected

  * Support for hex only ``softdevice`` sysbuild domains.
  * Updated the SPDX License List database to version 3.28.0.

* :ref:`nrf_profiler_script` script:

  * Updated:

>     * Fixed the ``ms_per_timestamp_tick`` parameter (defined in the :file: `rtt_nordic_config.py`) to accurately align with the system clock frequency of nRF52, nRF53, and nRF91 SoCs.

Reviewer 2: `... with the system clock's hardware timer frequency...`

  * Support for hex only ``softdevice`` sysbuild domains.
  * Updated the SPDX License List database to version 3.28.0.

* :ref:`nrf_profiler_script` script:

  * Updated:

    * Updated the scripts to use hardware clock information from SoC to enable accurate conversion from hardware clock cycles to seconds.

>       Refactored the parser to support the new RTT communication layout, including the handling of ``<ev_info_...>`` and ``<sys_config_...>`` data markers.

Reviewer 3: ```suggestion
  * Updated the scripts to use hardware clock information from SoC to enable accurate conversion from hardware clock cycles to seconds.
    Refactored the parser to support the new RTT communication layout, including the handling of ``<ev_info_...>`` and ``<sys_config_...>`` data markers.
```

Author: Corrected

  * Support for hex only ``softdevice`` sysbuild domains.
  * Updated the SPDX License List database to version 3.28.0.

* :ref:`nrf_profiler_script` script:

  * Updated the scripts to use hardware clock information from SoC to enable accurate conversion from hardware clock cycles to seconds.
    Refactored the parser to support the new RTT communication layout, including the handling of ``<ev_info_...>`` and ``<sys_config_...>`` data markers.

>   * Fixed the ``ms_per_timestamp_tick`` parameter (defined in the :file: `rtt_nordic_config.py`) to accurately align with the system clock's frequency of nRF52, nRF53, and nRF91 SoCs.

Reviewer 3: ```suggestion
  * Fixed the ``ms_per_timestamp_tick`` parameter (defined in the :file: `rtt_nordic_config.py`) to accurately align with the system clock's frequency of the nRF52, nRF53, and nRF91 Series SoCs.
```

Author: Corrected

  * Updated:

>     * Updated the scripts to use hardware clock information from SoC and to enable accurate conversion from hardware clock cycles to seconds.

Reviewer 2: ```suggestion
    * Updated the scripts to use hardware clock information from SoC to enable accurate conversion from hardware clock cycles to seconds.
```
